test/integrationtests/voight_kampff/tools.py

```diff
@@ -121,18 +156,46 @@ def wait_for_dialog(bus, dialogs, context=None, timeout=None):
                        provided by context or 10 seconds
     """
     if context:
-        timeout = timeout or context.step_timeout
+        timeout_duration = timeout or context.step_timeout
     else:
-        timeout = timeout or TIMEOUT
-    start_time = time.monotonic()
-    while time.monotonic() < start_time + timeout:
+        timeout_duration = timeout or DEFAULT_TIMEOUT
+    wait_for_dialog_match(bus, dialogs, timeout_duration)
+
+
+def wait_for_dialog_match(bus, dialogs, timeout=DEFAULT_TIMEOUT):
+    """Match dialogs spoken to the specified list of expected dialogs.
+
+    Only one of the dialogs in the provided list need to match for this
+    check to be successful.
+
+    Args:
+        bus (InterceptAllBusClient): Bus instance to listen on
+        dialogs (list): list of acceptable dialogs
+        timeout (int): how long to wait for the message, defaults to timeout
+                       provided by context or 10 seconds
+
+    Returns:
+        A boolean indicating if a match was found and the list of "speak"
+        events found on the message bus during the matching process.
+    """
+    match_found = False
+    speak_messages = list()
+    timeout_time = time.monotonic() + timeout
+    while time.monotonic() < timeout_time:
         for message in bus.get_messages('speak'):
+            speak_messages.append(message)
             dialog = message.data.get('meta', {}).get('dialog')
+            print('dialog: ', dialog, '\tutterance: ', message.data['utterance'])
```

Comment:
Is this a debug print that should be removed or is it intended to be here? Looks like it could be a bit spammy

Reply:
It is a debug print.  Removed.